File: lib/data/province.js
module.exports = class Province {
  constructor (name, homeOf, isSupplyCenter) {
    this.name = name
    this.homeOf = homeOf
    this.isSupplyCenter = isSupplyCenter || false
  }

  toString () {
    if (this.isSupplyCenter) {
      return `${this.name}*`
    } else {
      return this.name.toString()
    }
  }
}

File: core/src/main/scala/diplomacy/board/Province.scala
package diplomacy.board

final case class Province[Power_ <: Power](
  name: Name, homeOf: Option[Power_], isSupplyCenter: Boolean
) {
  override def toString: String = this.isSupplyCenter match {
    case true => s"${this.name}*"
    case false => this.name.toString
  }
}

object Province {
  trait TypeHelper {
    type Power <: diplomacy.board.Power
    type Province = diplomacy.board.Province[Power]
  }
}
